pkg/provider/azure/azclients/azclients.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package azclients
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage"
+	storageservice "github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/service"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient"
+	"sigs.k8s.io/cloud-provider-azure/pkg/azclient/configloader"
+)
+
+type Impl struct {
+	azureConfig *AzureConfig
+
+	storageAccountsClient    *armstorage.AccountsClient
+	blobContainersClient     *armstorage.BlobContainersClient
+	managementPoliciesClient *armstorage.ManagementPoliciesClient
+	blobServiceClient        *storageservice.Client
+}
+
+func (azclients *Impl) getTokenCredential() (azcore.TokenCredential, error) {
+	// Create token credential out of cloud credential config.
+	authProvider, err := azclient.NewAuthProvider(&azclients.azureConfig.ARMClientConfig, &azclients.azureConfig.AzureAuthConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create AuthProvider, %w", err)
```

Comment:
For consistency's sake, colons are better for error wrapping. Also, repeat for all instances.

```suggestion
		return nil, fmt.Errorf("failed to create AuthProvider: %w", err)
```

Reply:
Will do the replacement whereever it occurs.

Reply:
Done